### PR TITLE
V1.1.0 - Feat/ci

### PR DESCRIPTION
**Pull request by ttm02**
- Introduce CI that runs the 0-level benchmakrs and automatically updates the corrbench-dashboard
- use new release of MUST for the dashboard
- includes some minor fixes to the plot generation:
   - ability to specify the output format as an argument
   - read `MPI_CORRECTNESS_BM_DIR` from environment instead of require it as a cmd line arg

### CODE REVIEW

File: .github/workflows/visualize.yml [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    runs-on: ubuntu-latest
+    env:
+      MPI_CORRECTNESS_BM_DIR: ${{ github.workspace }}
+      MPI_CORRECTNESS_BM_EXPERIMENT_DIR: ${{ github.workspace }}/experiment
+    steps:
+      - uses: actions/checkout@v2
+# prerequisites for must
+      - run : sudo apt-get update && sudo apt-get install -y cmake git libxml2-dev openmpi-bin libopenmpi-dev
+      - run : sudo pip3 install numpy pandas lxml html5lib BeautifulSoup4
+# install MUST
```
Comment (jplehr) [inline .github/workflows/visualize.yml]: Should we remove these lines instead of commenting them out?

File: .github/workflows/visualize.yml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    runs-on: ubuntu-latest
+    env:
+      MPI_CORRECTNESS_BM_DIR: ${{ github.workspace }}
+      MPI_CORRECTNESS_BM_EXPERIMENT_DIR: ${{ github.workspace }}/experiment
+    steps:
+      - uses: actions/checkout@v2
+# prerequisites
+      - run : sudo apt-get update && sudo apt-get install -y cmake git libopenmpi-dev clang-9 llvm-9 llvm-9-dev
+# install PARCOACH
+# building parcoach is quite fast anyway
```
Comment (jplehr) [inline .github/workflows/visualize.yml]: Should we remove these lines instead of commenting them out?

File: .github/workflows/visualize.yml [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+#      - name: Cache PARCOACH
+#        id: cache-parcoach
+#        uses: actions/cache@v2
+#        with:
+#          path: parcoach
+#          key: ${{ runner.os }}-cache-parcoach-master
+      #TODO BUILD
```
Comment (jplehr) [inline .github/workflows/visualize.yml]: If that is a TODO, I would consider using the issues instead of in-code notes.
At least have the TODOs *also* as an issue in github.

File: .github/workflows/visualize.yml [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      - name: Build PARCOACH
+#        if: steps.cache-parcoach.outputs.cache-hit != 'true'
+        run: |
+           bash ./scripts/PARCOACH/install_tool.sh
+#TODO provide a container with the installation of MUST
```
Comment (jplehr) [inline .github/workflows/visualize.yml]: Should we have this TODO as an issue in the github issue tracker?

File: .github/workflows/visualize.yml [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+          name: results_MPI-Checker
+          path: ${{ github.workspace }}/experiment/MPI-Checker/results_MPI-Checker.json
+
+
+  generate-visualization:
+    needs: [run-mpi-checker, run-parcoach, run-must]
+    runs-on: ubuntu-latest
+    env:
+      MPI_CORRECTNESS_BM_DIR: ${{ github.workspace }}
+#TODO automatically get these version numbers
```
Comment (jplehr) [inline .github/workflows/visualize.yml]: This is the third, or so, TODO w.r.t. the CI configuration and setup.
Should we have one CI-related issue that lists the current TODOs?

File: README.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# MPI-CorrBench &middot; [![License](https://img.shields.io/badge/License-BSD%203--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause) 
+# MPI-CorrBench 1.1.0 &middot; [![License](https://img.shields.io/badge/License-BSD%203--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause)
```
Comment (jplehr) [inline README.md]: I personally would remove the dot, but do not have a strong opinion.

File: scripts/MUST/execute_tool_in_CI.sh [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+TIMEOUT_CMD="/usr/bin/timeout -k 120 120"
+
+# in $(pwd) ist eine datei testcase.c , die ausgewertet werden soll
```
Comment (jplehr) [inline scripts/MUST/execute_tool_in_CI.sh]: If the comment is relevant, we should have it in English instead of German.
Otherwise I would remove it.

File: scripts/MUST/install_tool.sh [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+cd MUST-v1.7
+mkdir build && cd build
+
+cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR ..
+make install
+
+cd ../..
+
+# check if installation was succesful
```
Comment (jplehr) [inline scripts/MUST/install_tool.sh]: Are these comments relevant / helpful?
If not, I would remove them, and add TODOs as issues in github.

File: scripts/PARCOACH/install_tool.sh [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# installs parcoach
+
+#PARCOACH_ROOT = $(pwd)/parcoach
```
Comment (jplehr) [inline scripts/PARCOACH/install_tool.sh]: This line seems irrelevant

**Review by jplehr — CHANGES_REQUESTED** [review]
I think we should remove the out-commented lines if they do not add to the understanding.
Otherwise seems fine.

File: .github/workflows/visualize.yml (hunk 1 above)
Comment (ttm02) [reply .github/workflows/visualize.yml]: ?? they arent commented out it is yust a comment, what these lines do

File: .github/workflows/visualize.yml (hunk 3 above)
Comment (ttm02) [reply .github/workflows/visualize.yml]: this is a leftover todo it is solved by now, I'll remove it

File: .github/workflows/visualize.yml (hunk 4 above)
Comment (ttm02) [reply .github/workflows/visualize.yml]: If we want container with the tools installed, we should add it to the issues - I'll remove it from the file

File: .github/workflows/visualize.yml (hunk 5 above)
Comment (ttm02) [reply .github/workflows/visualize.yml]: yes this should be listed in the issues section

File: .github/workflows/visualize.yml (hunk 1 above)
Comment (jplehr) [reply .github/workflows/visualize.yml]: The placement of the commend seems weird. I mean those below. 20 - 25

Comment (ttm02) [reply .github/workflows/visualize.yml]: do we want the caching? as must runs more than 4h (need for oversubscribe) I think that it is not important if we spend a bit of time building it - Think we can remove it

File: scripts/MUST/execute_tool_in_CI.sh (hunk 7 above)
Comment (ttm02) [reply scripts/MUST/execute_tool_in_CI.sh]: correct

File: README.md (hunk 6 above)
Comment (ttm02) [reply README.md]: Do we want to add a CI badge for the visualization (maybe with the embedded link to the dashboard)?

Comment (jplehr) [reply README.md]: That's a cool idea, but I would wait for it until the dashboard is online.

**Review by jplehr — APPROVED** [review]
Looks good to me.
